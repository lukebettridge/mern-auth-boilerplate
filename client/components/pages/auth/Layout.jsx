import React from "react";
import PropTypes from "prop-types";

import { Container, Heading, Paragraph } from "../../styles";

const Layout = props => {
	return (
		<Container small>
			<Heading>Webapp</Heading>
			{props.children}
			<Paragraph center light>
				Developed by{" "}
				<a
					href="https://paddl.co.uk/"
					rel="noopener noreferrer"
					target="_blank"
				>
					Paddl
				</a>
				<br />. A limited company registered in England and Wales. Registered
				number: 11992253.
			</Paragraph>
		</Container>
	);
};

Layout.propTypes = {
	children: PropTypes.any
};

export default Layout;
