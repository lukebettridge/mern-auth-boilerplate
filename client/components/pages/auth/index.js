import Login from "./login";
import Logout from "./logout";
import Register from "./register";
import ResetPassword from "./reset-password";

export { Login, Logout, Register, ResetPassword };
