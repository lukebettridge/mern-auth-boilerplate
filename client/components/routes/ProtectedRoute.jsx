import React from "react";
import PropType from "prop-types";
import { Route, Redirect } from "react-router-dom";
import Cookies from "js-cookie";

const ProtectedRoute = ({ component: Component, ...rest }) => {
	const isAuthenticated = () => {
		const authenticated = Cookies.get("authenticated");
		return authenticated && authenticated === "true";
	};

	return (
		<Route
			{...rest}
			render={props =>
				isAuthenticated() ? (
					<Component {...props} />
				) : (
					<Redirect
						to={{
							pathname: "/login",
							state: { from: props.location }
						}}
					/>
				)
			}
		/>
	);
};

ProtectedRoute.propTypes = {
	component: PropType.any,
	location: PropType.string
};

export default ProtectedRoute;
