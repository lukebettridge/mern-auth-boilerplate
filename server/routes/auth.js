const express = require("express");
const bcrypt = require("bcryptjs");
const jwt = require("jsonwebtoken");

const { mail, passwordHash, randomString } = require("../utils");

const validate = {
	register: require("../validation/register"),
	login: require("../validation/login"),
	resetPassword: require("../validation/reset-password")
};
const User = require("../src/models/User");
const RESET_KEY_STRING_LENGTH = 10;

const router = express.Router();

router.post("/register", (req, res) => {
	const { errors, isValid } = validate.register(req.body);
	if (!isValid) return res.status(400).json(errors);

	User.findOne({ email: req.body.email }).then(user => {
		if (user)
			return res.status(400).json({
				email:
					"The email address that you've entered is already associated with an account"
			});

		const newUser = new User({
			forename: req.body.forename,
			surname: req.body.surname,
			email: req.body.email,
			password: req.body.password
		});

		passwordHash(newUser.password, (err, hash) => {
			if (err) throw err;
			newUser.password = hash;
			newUser
				.save()
				.then(user => res.json(user))
				.catch(err => res.status(400).json(err));
		});
	});
});

router.post("/login", (req, res) => {
	const { errors, isValid } = validate.login(req.body);
	if (!isValid) return res.status(400).json(errors);

	const { email, password } = req.body;
	User.findOne({ email }).then(user => {
		if (!user)
			return res.status(404).json({
				email: "The email address that you've entered doesn't match any account"
			});

		bcrypt.compare(password, user.password).then(isMatch => {
			if (isMatch) {
				jwt.sign(
					{
						id: user.id,
						email: user.email
					},
					process.env.SECRET_OR_KEY,
					{
						expiresIn: "1d"
					},
					(_, token) => {
						res.cookie("jwt", token, {
							httpOnly: true,
							secure: process.env.NODE_ENV === "production"
						});
						res.json({ success: true });
					}
				);
			} else
				return res
					.status(400)
					.json({ password: "The password that you've entered is incorrect" });
		});
	});
});

router.get("/reset-password", (req, res) => {
	const { email } = req.query;
	if (!email)
		return res.status(400).json({
			email: "Email field is required"
		});

	User.findOne({ email }).then(user => {
		if (!user)
			return res.status(404).json({
				email: "The email address that you've entered doesn't match any account"
			});

		const date = new Date();
		date.setDate(date.getDate() + process.env.EXPIRE_PASSWORD_RESET_DAYS || 1);
		const resetKey = randomString(RESET_KEY_STRING_LENGTH, date);
		user
			.update({ resetKey })
			.then(() => {
				const url = `${process.env.BASE_URL}/auth/reset-password/${resetKey}`;
				mail({
					to: email,
					subject: `${process.env.APP_NAME} - Your password reset request`,
					text: `Hello ${user.forename},
					
					So, you've forgotten your password... no biggie, follow this link to set a new one: ${url}
					
					Thanks,
					The ${process.env.APP_NAME} Team`
				})
					.then(() => {
						res.json({ success: true });
					})
					.catch(err => res.status(400).json(err));
			})
			.catch(err => res.status(400).json(err));
	});
});

router.post("/reset-password", (req, res) => {
	const { errors, isValid } = validate.resetPassword(req.body);
	if (!isValid) return res.status(400).json(errors);

	const { newPassword, resetKey } = req.body;
	if (!resetKey)
		return res.status(400).json({
			resetKey: "No reset key was specified"
		});

	User.findOne({ resetKey }).then(user => {
		if (!user)
			return res.status(404).json({
				resetKey: "The reset key given doesn't match any account"
			});

		const milliseconds = Number(resetKey.substr(RESET_KEY_STRING_LENGTH));
		const date = new Date(milliseconds);
		if (new Date() > date)
			return res.status(400).json({
				resetKey: "The reset key given has expired"
			});

		passwordHash(newPassword, (err, hash) => {
			if (err) throw err;
			user
				.update({ password: hash, resetKey: undefined })
				.then(() => res.json({ success: true }))
				.catch(err => res.status(400).json(err));
		});
	});
});

module.exports = router;
