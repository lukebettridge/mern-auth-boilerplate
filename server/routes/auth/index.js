const router = require("express").Router();

router.post("/register", require("./register"));
router.post("/login", require("./login"));
router.post("/reset-password", require("./reset-password").post);
router.get("/reset-password", require("./reset-password").get);

module.exports = router;
