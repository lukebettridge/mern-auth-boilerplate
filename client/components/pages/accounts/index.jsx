import React, { useRef, useState } from "react";
import { Query } from "react-apollo";
import { gql } from "apollo-boost";

import Layout from "../layout";
import {
	FilterBody,
	FilterHeader,
	FilterWrap,
	Subheading,
	Status,
	Table
} from "components/styles";
import Input from "components/form/input";
import Button from "components/form/button";

const Accounts = () => {
	const queryInput = useRef();
	const [state, setState] = useState({ query: "" });

	const onSubmit = () =>
		setState({
			...state,
			query: queryInput.current.value
		});

	return (
		<Layout>
			<Subheading>Accounts</Subheading>

			<FilterWrap>
				<FilterHeader>
					<small>Search user accounts:</small>
				</FilterHeader>
				<FilterBody>
					<Input forwardRef={queryInput} mb="none" placeholder="Search Query" />
					<Button inline onClick={onSubmit}>
						Search
					</Button>
				</FilterBody>
			</FilterWrap>

			<Query
				fetchPolicy={"no-cache"}
				query={gql`
					{
						users(query: "${state.query}") {
							id
							forename
							surname
							email
							roles
							active
						}
					}
				`}
			>
				{({ loading, error, data }) => {
					if (loading) return <p>Loading...</p>;
					if (error) {
						return <p>error</p>;
					}
					return (
						<Table>
							<thead>
								<tr>
									<th className="col-1">Status</th>
									<th className="col-2">Surname</th>
									<th className="col-2">Forename</th>
									<th className="col-4">Email Address</th>
								</tr>
							</thead>
							<tbody>
								{data.users.map(user => (
									<tr key={user.id}>
										<td className="col-1">
											<Status success={user.active}>
												{user.active ? "Active" : "Inactive"}
											</Status>
										</td>
										<td className="col-2">{user.surname}</td>
										<td className="col-2">{user.forename}</td>
										<td className="col-4">{user.email}</td>
									</tr>
								))}
							</tbody>
						</Table>
					);
				}}
			</Query>
		</Layout>
	);
};

export default Accounts;
