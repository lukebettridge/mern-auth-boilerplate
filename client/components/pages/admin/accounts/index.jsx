import React, { useRef, useState } from "react";
import MediaQuery from "react-responsive";
import { Query } from "react-apollo";
import { gql } from "apollo-boost";

import Layout from "components/pages/layout";
import {
	FilterBody,
	FilterHeader,
	FilterWrap,
	Subheading,
	Status,
	Table,
	TableAction
} from "components/styles";
import breakpoints from "components/styles/breakpoints";
import Input from "components/form/input";
import Button from "components/form/button";
import AccountModal from "./AccountModal";

const Accounts = () => {
	const queryInput = useRef();
	const [state, setState] = useState({
		modalIsOpen: false,
		query: "",
		user: null
	});

	const onClick = user => {
		setState({ ...state, modalIsOpen: true, user });
	};

	const onSubmit = e => {
		e.preventDefault();
		setState({
			...state,
			query: queryInput.current.value
		});
	};

	return (
		<React.Fragment>
			<Layout>
				<Subheading>Accounts</Subheading>

				<FilterWrap>
					<FilterHeader>
						<small>Filter your results</small>
					</FilterHeader>
					<form noValidate onSubmit={onSubmit}>
						<FilterBody>
							<Input
								forwardRef={queryInput}
								mb="none"
								placeholder="Search..."
								secondary
							/>
							<Button inline secondary type="submit">
								Submit
							</Button>
						</FilterBody>
					</form>
				</FilterWrap>

				<Query
					fetchPolicy={"no-cache"}
					query={gql`
					{
						users(query: "${state.query}") {
							id
							forename
							surname
							email
							roles
							active
						}
					}
				`}
				>
					{({ loading, error, data, refetch }) => {
						if (loading) return <p>Loading...</p>;
						if (error) {
							return <p>error</p>;
						}
						return (
							<React.Fragment>
								<MediaQuery minWidth={breakpoints.l}>
									{matches =>
										matches ? (
											<Table>
												<thead>
													<tr>
														<th className="col-1">Status</th>
														<th className="col-2">Surname</th>
														<th className="col-2">Forename</th>
														<th className="col-4">Email Address</th>
													</tr>
												</thead>
												<tbody>
													{data.users.map(user => (
														<tr key={user.id} onClick={() => onClick(user)}>
															<td className="col-1">
																<Status success={user.active}>
																	{user.active ? "Active" : "Inactive"}
																</Status>
															</td>
															<td className="col-2">{user.surname}</td>
															<td className="col-2">{user.forename}</td>
															<td className="col-4">{user.email}</td>
														</tr>
													))}
												</tbody>
											</Table>
										) : (
											data.users.map(user => (
												<React.Fragment key={user.id}>
													<Table>
														<tbody>
															<tr>
																<th>Status</th>
																<td>
																	<Status success={user.active}>
																		{user.active ? "Active" : "Inactive"}
																	</Status>
																</td>
															</tr>
															<tr>
																<th>Surname</th>
																<td>{user.surname}</td>
															</tr>
															<tr>
																<th>Forename</th>
																<td>{user.forename}</td>
															</tr>
															<tr>
																<th>Email</th>
																<td>{user.email}</td>
															</tr>
														</tbody>
													</Table>
													<TableAction onClick={() => onClick(user)}>
														More Details
													</TableAction>
												</React.Fragment>
											))
										)
									}
								</MediaQuery>
								<AccountModal
									close={() => {
										refetch();
										setState({ ...state, modalIsOpen: false });
									}}
									isOpen={state.modalIsOpen}
									user={state.user}
								/>
							</React.Fragment>
						);
					}}
				</Query>
			</Layout>
		</React.Fragment>
	);
};

export default Accounts;
