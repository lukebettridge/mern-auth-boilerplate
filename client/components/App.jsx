import React, { useRef } from "react";
import { ApolloClient } from "apollo-client";
import { createHttpLink } from "apollo-link-http";
import { ApolloProvider } from "react-apollo";
import { InMemoryCache } from "apollo-cache-inmemory";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import NotificationSystem from "react-notification-system";

import { notification } from "utils";

import Context from "./context";

import ScrollToTop from "./routes/ScrollToTop";
import ProtectedRoute from "./routes/ProtectedRoute";

import GlobalStyle from "./styles/global";

import { Login, Logout, Register, ResetPassword } from "./pages/auth";
import { Accounts } from "./pages/admin";
import Home from "./pages/home";

const httpLink = createHttpLink({
	uri: `${baseURL}/graphql`,
	credentials: "include"
});

const client = new ApolloClient({
	link: httpLink,
	cache: new InMemoryCache(),
	credentials: "include"
});

const App = () => {
	const notificationSystem = useRef();
	const defaultContext = {
		notification: notification(notificationSystem)
	};

	return (
		<ApolloProvider client={client}>
			<Router>
				<ScrollToTop />
				<GlobalStyle />
				<Context.Provider value={defaultContext}>
					<Switch>
						<Route component={Login} path="/auth/login" />
						<Route component={Logout} path="/auth/logout" />
						<Route component={Register} path="/auth/register" />
						<Route
							component={ResetPassword}
							path="/auth/reset-password/:resetKey"
						/>
						<ProtectedRoute component={Home} path="/(|home)/" />
						<ProtectedRoute component={Accounts} path="/admin/accounts" />
					</Switch>
					<NotificationSystem
						ref={notificationSystem}
						style={{
							NotificationItem: {
								DefaultStyle: {
									border: "none",
									fontWeight: "200",
									padding: "18px"
								}
							},
							Title: {
								DefaultStyle: { fontSize: "18px" }
							},
							Dismiss: {
								DefaultStyle: { display: "none" }
							}
						}}
					/>
				</Context.Provider>
			</Router>
		</ApolloProvider>
	);
};

export default App;
