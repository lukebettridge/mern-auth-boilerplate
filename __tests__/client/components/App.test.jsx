import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import { ApolloProvider } from "react-apollo";
import { Route } from "react-router-dom";

import ProtectedRoute from "components/routes/ProtectedRoute";
import { Login, Logout, Register, ResetPassword } from "components/pages/auth";
import { Accounts } from "components/pages/admin";
import Home from "components/pages/home";
import App from "components/App";

describe("App component", () => {
	it("snapshot renders", () => {
		const subject = shallow(<App />);

		expect(toJSON(subject)).toMatchSnapshot();
	});

	it("renders apollo provider with client", () => {
		const subject = shallow(<App />);
		const apolloProvider = subject.find(ApolloProvider);

		expect(apolloProvider.length).toEqual(1);
		expect(apolloProvider.get(0).props.client).toBeTruthy();
	});

	it("renders all routes", () => {
		const subject = shallow(<App />);
		expect(subject.find(Route).length).toEqual(4);
		expect(subject.find(ProtectedRoute).length).toEqual(2);
	});

	[
		{
			component: Login,
			route: "/auth/login"
		},
		{
			component: Logout,
			route: "/auth/logout"
		},
		{
			component: Register,
			route: "/auth/register"
		},
		{
			component: ResetPassword,
			route: "/auth/reset-password/:resetKey"
		}
	].map(({ component, route }, index) => {
		it(`passes route props for ${route}`, () => {
			const subject = shallow(<App />);
			const props = subject.find(Route).get(index).props;

			expect(props.component).toBe(component);
			expect(props.path).toEqual(route);
		});
	});

	[
		{
			component: Home,
			route: "/(|home)/"
		},
		{
			component: Accounts,
			route: "/admin/accounts"
		}
	].map(({ component, route }, index) => {
		it(`passes protected route props for ${route}`, () => {
			const subject = shallow(<App />);
			const props = subject.find(ProtectedRoute).get(index).props;

			expect(props.component).toBe(component);
			expect(props.path).toEqual(route);
		});
	});
});
