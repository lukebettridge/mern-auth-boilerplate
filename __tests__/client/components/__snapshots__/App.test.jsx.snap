// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot renders 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "cache": InMemoryCache {
        "addTypename": true,
        "cacheKeyRoot": KeyTrie {
          "weakness": true,
        },
        "config": Object {
          "addTypename": true,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "freezeResults": false,
          "resultCaching": true,
        },
        "data": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "maybeBroadcastWatch": [Function],
        "optimisticData": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "silenceBroadcast": false,
        "storeReader": StoreReader {
          "executeSelectionSet": [Function],
          "executeStoreQuery": [Function],
          "executeSubSelectedArray": [Function],
          "freezeResults": false,
        },
        "storeWriter": StoreWriter {},
        "typenameDocumentCache": Map {},
        "watches": Set {},
      },
      "clearStoreCallbacks": Array [],
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": ApolloLink {
        "request": [Function],
      },
      "localState": LocalState {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
        "client": [Circular],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "queryManager": QueryManager {
        "assumeImmutableResults": false,
        "clientAwareness": Object {
          "name": undefined,
          "version": undefined,
        },
        "dataStore": DataStore {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
        },
        "fetchQueryRejectFns": Map {},
        "idCounter": 1,
        "inFlightLinkObservables": Map {},
        "link": ApolloLink {
          "request": [Function],
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": true,
            "cacheKeyRoot": KeyTrie {
              "weakness": true,
            },
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {},
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {},
          },
          "client": [Circular],
        },
        "mutationStore": MutationStore {
          "store": Object {},
        },
        "onBroadcast": [Function],
        "pollingInfoByQueryId": Map {},
        "queries": Map {},
        "queryDeduplication": true,
        "queryStore": QueryStore {
          "store": Object {},
        },
        "ssrMode": false,
        "transformCache": WeakMap {},
      },
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": KeyTrie {
            "weakness": true,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
      },
      "typeDefs": undefined,
      "version": "2.6.8",
      "watchQuery": [Function],
    }
  }
>
  <BrowserRouter>
    <withRouter(ScrollToTop) />
    <GlobalStyleComponent />
    <ContextProvider
      value={
        Object {
          "notification": Object {
            "ref": Object {
              "current": undefined,
            },
            "success": [Function],
          },
        }
      }
    >
      <Switch>
        <Route
          component={[Function]}
          path="/auth/login"
        />
        <Route
          component={[Function]}
          path="/auth/logout"
        />
        <Route
          component={[Function]}
          path="/auth/register"
        />
        <Route
          component={[Function]}
          path="/auth/reset-password/:resetKey"
        />
        <ProtectedRoute
          component={[Function]}
          path="/(|home)/"
        />
        <ProtectedRoute
          component={[Function]}
          path="/admin/accounts"
          roles={
            Array [
              "admin",
            ]
          }
        />
      </Switch>
      <NotificationSystem
        allowHTML={false}
        noAnimation={false}
        style={
          Object {
            "Dismiss": Object {
              "DefaultStyle": Object {
                "display": "none",
              },
            },
            "NotificationItem": Object {
              "DefaultStyle": Object {
                "border": "none",
                "fontWeight": "200",
                "padding": "18px",
              },
            },
            "Title": Object {
              "DefaultStyle": Object {
                "fontSize": "18px",
              },
            },
          }
        }
      />
    </ContextProvider>
  </BrowserRouter>
</ApolloProvider>
`;
